var/spack/repos/builtin/packages/postgresql/package.py

```diff
@@ -125,3 +125,32 @@ def setup_dependent_run_environment(self, env, dependent_spec):
             env.prepend_path('TCLLIBPATH', self.prefix.lib)
         if '+python' in spec:
             env.prepend_path('PYTHONPATH', self.prefix.lib)
+
+    @property
+    def libs(self):
+        dyn_libs = ['libecpg_compat', 'libecpg', 'libpgtypes', 'libpq',
+                    'libpqwalreceiver', 'plpgsql', 'pgoutput.so']
+        fl_dyn = find_libraries(dyn_libs, self.prefix, shared=True,
+                                recursive=True)
+        stat_libs = ['libecpg_compat', 'libecpg', 'libpgcommon',
+                     'libpgcommon_shlib', 'libpgfeutils', 'libpgport',
+                     'libpgport_shlib', 'libpgtypes', 'libpq']
+        fl_stat = find_libraries(stat_libs, self.prefix, shared=False,
+                                 recursive=True)
+        if '+perl' in self.spec:
```

Comment:
Probably much simpler to append to `dyn_libs` and `stat_libs`, then you only have to call `find_libraries` twice.

Reply:
I thought I have to do so since for dynamic libraries `Shared=True` and for static `Shared=False`?

Reply:
Yes, you'll still need to call `find_libraries` twice (once for shared and once for static) but you don't need to call it five times like you currently do.